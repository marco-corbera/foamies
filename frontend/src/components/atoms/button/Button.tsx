import React from "react";

type ButtonProps = {
  children: React.ReactNode;
  onClick?: () => void;
  variant?: "primary" | "secondary";
  disabled?: boolean;
  className?: string;
};

const Button: React.FC<ButtonProps> = ({
  children,
  onClick,
  variant = "primary",
  disabled = false,
  className = "",
}) => {
  const baseClasses =
    "px-4 py-2 rounded-full text-white font-semibold transition duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2";
  const variantClasses =
    variant === "primary"
      ? "bg-primary hover:bg-blue-600 focus:ring-blue-400"
      : "bg-secondary hover:bg-gray-600 focus:ring-gray-400";
  const disabledClasses = disabled
    ? "bg-gray-300 text-gray-700 cursor-not-allowed"
    : "";

  return (
    <button
      onClick={onClick}
      className={`${baseClasses} ${variantClasses} ${disabledClasses} ${className}`}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

export default Button;
