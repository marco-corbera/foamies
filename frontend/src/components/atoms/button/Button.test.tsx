import { render, screen, waitFor } from "@testing-library/react";

import Button from "./Button";

describe("Button tests", () => {
  it("Matches snapshot", () => {
    const { asFragment } = render(<Button children={"My great button"} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the button", () => {
    render(<Button children={"My great button"} />);
    expect(screen.getByRole("button")).toBeInTheDocument();
  });
});
