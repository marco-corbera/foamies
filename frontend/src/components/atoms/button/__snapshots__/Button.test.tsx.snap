// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-full text-white font-semibold transition duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 bg-primary hover:bg-blue-600 focus:ring-blue-400 "
  >
    My great button
  </button>
</DocumentFragment>
`;
